internal/controller/docs/cluster_provider.go

```diff
@@ -42,20 +46,23 @@ func (d *ClusterProvider) EnsureDocsTopic(addon *internal.Addon) error {
 		},
 		Spec: v1alpha1.ClusterDocsTopicSpec{CommonDocsTopicSpec: addon.Docs[0].Template},
 	}
-
 	d.log.Infof("- ensuring ClusterDocsTopic %s", addon.ID)
-	err := d.dynamicClient.Create(context.Background(), cdt)
-	switch {
-	case err == nil:
-	case apiErrors.IsAlreadyExists(err):
-		if err := d.updateClusterDocsTopic(addon); err != nil {
-			return errors.Wrapf(err, "while ClusterDocsTopic %s already exists", addon.ID)
-		}
-	default:
-		return errors.Wrapf(err, "while creating ClusterDocsTopic %s", addon.ID)
-	}
 
-	return nil
+	return wait.PollImmediate(time.Second*500, time.Second*3, func() (bool, error) {
```

Comment:
Interval 500sec and timeout 3sek? It shouldn't be the other way around or the first should be milliseconds?

Reply:
Changed all the occurrences to 500milliseconds